internal/services/storage/storage_account_resource.go

```
@@ -1096,6 +1108,10 @@ func resourceStorageAccountCreate(d *pluginsdk.ResourceData, meta interface{}) e
 		},
 	}
 
+	if accountKind == string(storage.KindStorageV2) {
+		parameters.Encryption.RequireInfrastructureEncryption = &infrastructureEncryption
+	}
```

we should probably return an error if the user is trying to set this with v1?